src/coreclr/jit/valuenum.h

```diff
@@ -1527,6 +1650,110 @@ FORCEINLINE T ValueNumStore::SafeGetConstantValue(Chunk* c, unsigned offset)
     }
 }
 
+#if defined(FEATURE_SIMD)
+template <>
+FORCEINLINE simd8_t ValueNumStore::SafeGetConstantValue<simd8_t>(Chunk* c, unsigned offset)
+{
+    if (c->m_typ == TYP_SIMD8)
+    {
+        return reinterpret_cast<VarTypConv<TYP_SIMD8>::Lang*>(c->m_defs)[offset];
+    }
+    else
+    {
+        assert(false);
+        return {};
+    }
```

Comment:
This pattern seems like quite the verbose way of saying:
```
assert(c->m_typ == TYP_SIMD);
return reinterpret_cast<VarTypConv<TYP_SIMD>::Lang*>(c->m_defs)[offset];
```
I don't think we need to return zero in release build in case something calls this with a wrong type, it will be broken either way.

Reply:
Was just matching the existing behavior here, which is to return zero in the case of failure. Am happy to update if we think its better

Reply:
Yes, I think it would be better to make the code more compact for these specializations.

Reply:
Fixed.